### PR TITLE
feat: SSSS (Server Specific Settings System)

### PR DESCRIPTION
**Pull request by VALERA771**
## Description
**Describe the changes** 
Adding a wrappers for Server Specific Settings system

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?** (if this is a feature change)

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

**Other information**:
As for now, changes are partly tested

<br />

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentations
<br />

## Submission checklist

- [x] I have checked the project can be compiled
- [x] I have tested my changes and it worked as expected

### Patches (if there are any changes related to Harmony patches)
- [x] I have checked no IL patching errors in the console

### Other
- [ ] Still requires more testing

### CODE REVIEW

**Review by BoltonDev — CHANGES_REQUESTED** [review]
Make wrappers implement IWrapper

**Comment by VALERA771** [comment]
IWrapper interface requires type to inherit from MonoBehaviour. Idk why I can remove it but... I think it was made for a reason

**Comment by BoltonDev** [comment]
> IWrapper interface requires type to inherit from MonoBehaviour. Idk why, I can remove it but... I think it was made for a reason

I think we can remove it because there is no reason. That (https://github.com/Exiled-Team/EXILED/pull/1495) was the PR that added IWrapper, and actually with IWrapper, there is nothing that requires type to inherit from MonoBehaviour

**Comment by VALERA771** [comment]
I think there is requirement for it cuz IWrapper was made only for classes that wraps base game objects (that usually a mono behavior)

**Comment by VALERA771** [comment]
But ok I'll remove it

**Comment by JayXTQ** [comment]
Just curious about this whilst reading it, how would I add the options? would I just do "new Option" with the option values, and then SettingsBase.Create or SettingsBase.Register? What would be the plan here?

Also should this be done on OnEnabled or? I think more documentation should be done for this

**Comment by VALERA771** [comment]
Register method takes collection of setting base as a parameter so I think it's easy to understand how to register them.
On how creating settings more docs can be created to explain how everything works

**Comment by JayXTQ** [comment]
> Register method takes collection of setting base as a parameter so I think it's easy to understand how to register them. On how creating settings more docs can be created to explain how everything works

Yeah there needs to be more documentation, like what's the difference between register and create and how do we actually put our options inside of the server specific settings menu for all of our players. I really need this for my server so this will be really useful to understand.

**Comment by VALERA771** [comment]
@JayXTQ i've added some docs to method, Waiting for your feedback

**Comment by JayXTQ** [comment]
Yeah it looks alright considering I have delved into SSS on the pluginapi level, maybe add some more docs to say what the normal process is, like create a new item from one of the classes and then push that into register or send to new player, etc

**Review by BoltonDev — APPROVED** [review]

File: EXILED/Exiled.API/Interfaces/IWrapper.cs [hunk 1]
```diff
@@ -7,14 +7,11 @@
...
 namespace Exiled.API.Interfaces
 {
-    using UnityEngine;
-
     /// <summary>
     /// Defines the contract for classes that wrap a base-game object.
     /// </summary>
     /// <typeparam name="T">The base-game class that is being wrapped.</typeparam>
     public interface IWrapper<T>
-        where T : MonoBehaviour
```
Comment (louis1706) [inline EXILED/Exiled.API/Interfaces/IWrapper.cs:17]: why ?

**Review by louis1706 — CHANGES_REQUESTED** [review]

File: EXILED/Exiled.API/Interfaces/IWrapper.cs (hunk 1 above)
Comment (louis1706) [reply EXILED/Exiled.API/Interfaces/IWrapper.cs:17]: oh sorry i didn't read before i accept let me ask of hwy @Thundermaker300  made this like that

File: EXILED/Exiled.API/Features/Core/UserSettings/TwoButtonsSetting.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        public bool IsFirst
+        {
+            get => Base.SyncIsA;
+            set => Base.SyncIsB = !value;
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the second option is default.
+        /// </summary>
+        public bool IsSecondDefalut
```
Comment (Misfiy) [inline EXILED/Exiled.API/Features/Core/UserSettings/TwoButtonsSetting.cs:78]: IsSecondDefault

File: EXILED/Exiled.API/Features/Core/UserSettings/ButtonSetting.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        /// </summary>
+        /// <param name="id"><inheritdoc cref="SettingBase.Id"/></param>
+        /// <param name="label"><inheritdoc cref="SettingBase.Label"/></param>
+        /// <param name="buttonText"><inheritdoc cref="Text"/></param>
+        /// <param name="holdTime"><inheritdoc cref="HoldTime"/></param>
+        /// <param name="hintDescription"><inheritdoc cref="SettingBase.HintDescription"/></param>
+        /// <param name="header"><inheritdoc cref="SettingBase.Header"/></param>
+        /// <param name="onChanged"><inheritdoc cref="SettingBase.OnChanged"/></param>
+        public ButtonSetting(int id, string label, string buttonText, float holdTime = 0.0f, string hintDescription = null, HeaderSetting header = null, Action<Player, SettingBase> onChanged = null)
+            : base(new SSButton(id, label, buttonText, holdTime, hintDescription), header, onChanged)
```
Comment (Misfiy) [inline EXILED/Exiled.API/Features/Core/UserSettings/ButtonSetting.cs:32]: Shouldnt this be :this()

Comment (VALERA771) [reply EXILED/Exiled.API/Features/Core/UserSettings/ButtonSetting.cs:32]: Why? It's cctor from SettingBase (`SettingBase(ServerSpecificSettingBase, HeaderSetting, Action)`

File: EXILED/Exiled.API/Features/Core/UserSettings/TwoButtonsSetting.cs (hunk 2 above)
Comment (VALERA771) [reply EXILED/Exiled.API/Features/Core/UserSettings/TwoButtonsSetting.cs:78]: Bruh

**Review by louis1706 — APPROVED** [review]

**Review by louis1706 — APPROVED** [review]

**Pull request merged**